In [ ]:
import pandas as pd
import sqlite3

# Load Excel file
df = pd.read_excel("Dataset_Cleaned.xlsx")

# Create SQL database in memory
conn = sqlite3.connect(':memory:')

# Load dataframe into SQL table
df.to_sql('orders', conn, index=False, if_exists='replace')

print("Database created successfully!")
print("Table name: orders")
print("Total rows:", len(df))

Database created successfully!
Table name: orders
Total rows: 1200


In [ ]:
query = """
SELECT *
FROM orders
LIMIT 10
"""
result = pd.read_sql_query(query, conn)
print(result)

     OrderID        Date CustomerID  Product  Quantity  UnitPrice  \
0  ORD200000  2023/01/04     C72649  Monitor         5     570.62   
1  ORD200001  2024/08/23     C75739    Phone         2     151.35   
2  ORD200002  2024/02/27     C81728   Tablet         5     550.68   
3  ORD200003  2023/10/15     C33540    Chair         1     273.19   
4  ORD200004  2025/05/08     C81840  Printer         4     626.01   
5  ORD200005  2023/10/23     C37249    Phone         2     245.86   
6  ORD200006  2025/06/17     C83492   Laptop         1     664.42   
7  ORD200007  2023/05/12     C41460  Monitor         5     149.55   
8  ORD200008  2025/04/02     C26817    Phone         2     134.28   
9  ORD200009  2023/11/21     C31946     Desk         4     509.38   

  ShippingAddress PaymentMethod OrderStatus TrackingNumber  ItemsInCart  \
0     928 Main St    Debit Card     Shipped    TRK37947903            7   
1     823 Main St        Online     Shipped    TRK91186779            3   
2     512 Main 

In [ ]:
query = """
SELECT COUNT(*) AS Total_Orders
FROM orders
"""
result = pd.read_sql_query(query, conn)
print(result)

   Total_Orders
0          1200


In [ ]:
query = """
SELECT SUM(TotalPrice) AS Total_Revenue
FROM orders
"""
result = pd.read_sql_query(query, conn)
print(result)

   Total_Revenue
0     1264761.96


In [ ]:
query = """
SELECT
    AVG(TotalPrice) AS Average_Order_Value,
    MIN(TotalPrice) AS Minimum_Order,
    MAX(TotalPrice) AS Maximum_Order
FROM orders
"""
result = pd.read_sql_query(query, conn)
print(result)

   Average_Order_Value  Minimum_Order  Maximum_Order
0            1053.9683          11.39         3456.4


In [ ]:
query = """
SELECT
    OrderStatus,
    COUNT(*) AS Number_of_Orders
FROM orders
GROUP BY OrderStatus
ORDER BY Number_of_Orders DESC
"""
result = pd.read_sql_query(query, conn)
print(result)

  OrderStatus  Number_of_Orders
0   Cancelled               250
1    Returned               247
2     Pending               237
3     Shipped               235
4   Delivered               231


In [ ]:
query = """
SELECT
    Product,
    COUNT(*) AS Total_Orders,
    SUM(TotalPrice) AS Total_Revenue,
    AVG(TotalPrice) AS Avg_Order_Value
FROM orders
GROUP BY Product
ORDER BY Total_Revenue DESC
"""
result = pd.read_sql_query(query, conn)
print(result)


   Product  Total_Orders  Total_Revenue  Avg_Order_Value
0    Chair           178      195620.11      1098.989382
1  Printer           181      195612.61      1080.732652
2   Laptop           173      192126.56      1110.558150
3   Tablet           179      186568.95      1042.284637
4  Monitor           163      175651.41      1077.616012
5     Desk           170      167459.93       985.058412
6    Phone           156      151722.39       972.579423


In [ ]:
query = """
SELECT
    OrderID,
    Product,
    TotalPrice,
    OrderStatus
FROM orders
WHERE OrderStatus = 'Cancelled'
ORDER BY TotalPrice DESC
LIMIT 10
"""
result = pd.read_sql_query(query, conn)
print(result)

     OrderID  Product  TotalPrice OrderStatus
0  ORD200469    Chair     3384.90   Cancelled
1  ORD200328   Tablet     3370.20   Cancelled
2  ORD200527    Chair     3267.35   Cancelled
3  ORD200768   Tablet     3267.30   Cancelled
4  ORD200889  Monitor     3253.60   Cancelled
5  ORD200802    Chair     3223.20   Cancelled
6  ORD200086  Printer     3215.15   Cancelled
7  ORD200296     Desk     3194.00   Cancelled
8  ORD200364    Phone     3143.70   Cancelled
9  ORD200764   Laptop     3137.15   Cancelled


In [ ]:
query = """
SELECT
    OrderID,
    Product,
    TotalPrice,
    PaymentMethod,
    OrderStatus
FROM orders
WHERE TotalPrice > 2000
ORDER BY TotalPrice DESC
"""
result = pd.read_sql_query(query, conn)
print(result)

       OrderID  Product  TotalPrice PaymentMethod OrderStatus
0    ORD200789   Tablet     3456.40        Online   Delivered
1    ORD201122  Monitor     3390.95        Online    Returned
2    ORD200632   Laptop     3390.80     Gift Card   Delivered
3    ORD200469    Chair     3384.90          Cash   Cancelled
4    ORD200328   Tablet     3370.20        Online   Cancelled
..         ...      ...         ...           ...         ...
175  ORD201035   Laptop     2024.44          Cash   Cancelled
176  ORD200181  Printer     2021.95        Online     Shipped
177  ORD200915   Laptop     2011.96    Debit Card     Shipped
178  ORD200979  Monitor     2007.18          Cash   Delivered
179  ORD201136   Tablet     2004.99     Gift Card     Pending

[180 rows x 5 columns]


In [ ]:
query = """
SELECT
    ReferralSource,
    COUNT(*) AS Total_Orders,
    SUM(TotalPrice) AS Total_Revenue,
    AVG(TotalPrice) AS Avg_Order_Value
FROM orders
GROUP BY ReferralSource
ORDER BY Total_Orders DESC
"""
result = pd.read_sql_query(query, conn)
print(result)

  ReferralSource  Total_Orders  Total_Revenue  Avg_Order_Value
0      Instagram           259      275285.45      1062.878185
1          Email           250      261808.55      1047.234200
2         Google           241      250441.48      1039.176266
3       Facebook           228      250410.90      1098.293421
4       Referral           222      226815.58      1021.691802


In [ ]:
query = """
SELECT
    PaymentMethod,
    COUNT(*) AS Total_Orders,
    SUM(TotalPrice) AS Total_Revenue
FROM orders
GROUP BY PaymentMethod
ORDER BY Total_Orders DESC
"""
result = pd.read_sql_query(query, conn)
print(result)

  PaymentMethod  Total_Orders  Total_Revenue
0        Online           258      262442.94
1          Cash           246      259786.29
2   Credit Card           234      263847.63
3    Debit Card           232      232361.18
4     Gift Card           230      246323.92


In [ ]:
query = """
SELECT
    SUBSTR(Date, 1, 7) AS Month,
    COUNT(*) AS Total_Orders,
    SUM(TotalPrice) AS Monthly_Revenue,
    AVG(TotalPrice) AS Avg_Order_Value
FROM orders
GROUP BY Month
ORDER BY Month ASC
"""
result = pd.read_sql_query(query, conn)
print(result)

      Month  Total_Orders  Monthly_Revenue  Avg_Order_Value
0   2023/01            47         56685.75      1206.079787
1   2023/02            37         40117.66      1084.261081
2   2023/03            43         48609.37      1130.450465
3   2023/04            31         27751.71       895.216452
4   2023/05            49         63836.84      1302.792653
5   2023/06            45         49500.19      1100.004222
6   2023/07            44         42820.66       973.196818
7   2023/08            51         54352.14      1065.728235
8   2023/09            29         29526.67      1018.161034
9   2023/10            47         52607.85      1119.315957
10  2023/11            41         43079.67      1050.723659
11  2023/12            46         43754.73       951.189783
12  2024/01            32         38528.08      1204.002500
13  2024/02            32         36909.57      1153.424063
14  2024/03            36         36030.90      1000.858333
15  2024/04            50         49613.

In [ ]:
query = """
SELECT
    Product,
    COUNT(*) AS Cancelled_Orders
FROM orders
WHERE OrderStatus = 'Cancelled'
GROUP BY Product
ORDER BY Cancelled_Orders DESC
"""
result = pd.read_sql_query(query, conn)
print(result)

   Product  Cancelled_Orders
0    Chair                45
1  Printer                35
2  Monitor                35
3   Laptop                35
4     Desk                35
5   Tablet                34
6    Phone                31


In [ ]:
query = """
SELECT
    COUNT(*) AS Total_Orders,
    SUM(TotalPrice) AS Total_Revenue,
    AVG(TotalPrice) AS Avg_Order_Value,
    MIN(TotalPrice) AS Min_Order,
    MAX(TotalPrice) AS Max_Order,
    COUNT(CASE WHEN OrderStatus = 'Cancelled' THEN 1 END) AS Cancelled_Orders,
    COUNT(CASE WHEN OrderStatus = 'Delivered' THEN 1 END) AS Delivered_Orders
FROM orders
"""
result = pd.read_sql_query(query, conn)
print(result)

   Total_Orders  Total_Revenue  Avg_Order_Value  Min_Order  Max_Order  \
0          1200     1264761.96        1053.9683      11.39     3456.4   

   Cancelled_Orders  Delivered_Orders  
0               250               231  


## SQL Analysis — Key Findings

1. Total Orders: 1,200
2. Total Revenue: $1,264,762
3. Average Order Value: $1,054
4. Highest Single Order: Check query result
5. Most Cancelled Product: Check query result
6. Top Referral Source: Instagram
7. Top Product by Revenue: Chair
8. Monthly trend shows revenue fluctuation
   with no consistent upward trend

## SQL Commands Used
- SELECT — to choose columns
- FROM — to specify table
- WHERE — to filter rows
- GROUP BY — to group categories
- ORDER BY — to sort results
- COUNT() — to count orders
- SUM() — to total revenue
- AVG() — to average values
- MIN() / MAX() — to find extremes